scri/flux.py

```diff
@@ -437,12 +429,298 @@ def angular_momentum_flux(h, hdot=None):
+
+    def mat_el_ethbar_chi(s, l3, m3, l1, m1, l2, m2):
+
+        cg1 = np.sqrt(2) * CG(l1, m1, l2, m2, l3, m3)
+        cg2 = CG(l1, 1.0, l2, -1 - s, l3, -s)
+
+        return np.sqrt((2.0 * l1 + 1.0) * (2.0 * l2 + 1.0) / (4.0 * np.pi * (2.0 * l3 + 1))) * cg1 * cg2
+
+    for ell in range(ell_min, ell_max + 1):
+        ellp_min = max(ell_min, ell - 1)
+        ellp_max = min(ell_max, ell + 1)
+        for ellp in range(ellp_min, ellp_max + 1):
+            for m in range(-ell, ell + 1):
+                mp = round(m + 1 * sign)
+                if (mp < -ellp) or (mp > ellp):
+                    continue
+                yield (ellp, mp, ell, m, (prefac * mat_el_ethbar_chi(s, ellp, mp, 1.0, sign, ell, m)))
+
+
+ethbar_chi_plus = functools.partial(ethbar_chi_plusminus, sign=+1)
+ethbar_chi_minus = functools.partial(ethbar_chi_plusminus, sign=-1)
+ethbar_chi_plus.__doc__ = ethbar_chi_plusminus.__doc__
+ethbar_chi_minus.__doc__ = ethbar_chi_plusminus.__doc__
+
+
+def boost_flux(h, hdot=None):
+    """p_z, p_plusminus and the matrix elements defined above will be used here.
+    matrix_expectation_value is used to calcuate the component of the flux.
+    The spin value taken by the matrix element is specified for each term.
+    """
+
+    from .waveform_modes import WaveformModes
+    from . import h as htype
+    from . import hdot as hdottype
+
+    if not isinstance(h, WaveformModes):
+        raise ValueError(
+            "Boost fluxes can only be calculated from a `WaveformModes` object; "
+            + "`h` is of type `{0}`.".format(type(h))
+        )
+    if (hdot is not None) and (not isinstance(hdot, WaveformModes)):
+        raise ValueError(
+            "Boost fluxes can only be calculated from a `WaveformModes` object; "
+            + "`hdot` is of type `{0}`.".format(type(hdot))
+        )
+    if h.dataType == htype:
+        if hdot is None:
+            hdot = h.copy()
+            hdot.dataType = hdottype
+            hdot.data = h.data_dot
+        elif hdot.dataType != hdottype:
+            raise ValueError(
+                "Input argument `hdot` is expected to have data of type `hdot`; "
+                + "this `hdot` waveform data has type `{0}`".format(h.data_type_string)
+            )
+    else:
+        raise ValueError(
+            "Input argument `h` is expected to have data of type `h`; "
+            + "this `h` waveform data has type `{0}`".format(h.data_type_string)
+        )
+
+    # h and h_dot are spin raised and lowered in the expression for boost flux.
+    # Hence, one needs to define new waveform like object from h and hdot that
+    # contains the updated data corresponding to the raising and lowering of the spin.
+    # Appropriate ladder factor needs to be multiplied to each (l,m) mode.
+    # eth and ethbar functions are used here for this operation.
+
+    # eth_h --> Spin of h raised by 1; s = -1
+    eth_h = h.copy()
+    eth_h.data = h.eth
```

Comment:
Another triple check on this convention. You are using the Newman-Penrose eth operator here, please make sure this is accounted for.